lib/bitarray.rb

```diff
@@ -49,7 +49,7 @@ def each_byte
   # Returns the total number of bits that are set
   # (The technique used here is about 6 times faster than using each or inject direct on the bitfield)
   def total_set
-    each_byte.inject(0) { |a, byte| a += byte & 1 and byte >>= 1 until byte == 0; a }
+    @field.bytes.inject(0) { |a, byte| a += byte & 1 and byte >>= 1 until byte == 0; a }
```

Comment:
I benchmarked this. The version using `each_byte` takes ~1.5x as long.

So, I reverted back to your implementation.